src/System.Data.Common/src/System/Data/Common/IDbColumnSchemaGenerator.cs

```diff
@@ -0,0 +1,11 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Data.Common
+{
+    public interface IDbColumnSchemaGenerator
```

Comment:
We brainstormed over both the ideas and decided to choose Generator. 
Provider resonated with other generic concepts. 
e.g https://msdn.microsoft.com/en-us/library/microsoft.data.schema.databaseschemaprovider(v=vs.100).aspx